From 0390ac046d1c86fe858bf566a5b3e2e9b4065a88 Mon Sep 17 00:00:00 2001
From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Mon, 23 Apr 2018 17:53:13 +0800
Subject: [PATCH] MetaShapedTexture: Disable mipmapping (emulation)

Window previews (e.g. in overview) were being mipmapped (antialiased)
in real time. This is generally considered a bad idea in OpenGL because
most desktop GPUs can't generate mipmaps (scaled and smoothed copies of
a texture) efficiently.

So turn it off, just as we did in Compiz many years ago.

The measured performance win by doing this is a 20% reduction in
mutter's render time. But the benefit to reduced GPU load could be even
better than that.
---
 src/compositor/meta-shaped-texture.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor/meta-shaped-texture.c b/src/compositor/meta-shaped-texture.c
index 98346c6ae..7e2b6df98 100644
--- a/src/compositor/meta-shaped-texture.c
+++ b/src/compositor/meta-shaped-texture.c
@@ -132,7 +132,7 @@ meta_shaped_texture_init (MetaShapedTexture *self)
 
   priv->texture = NULL;
   priv->mask_texture = NULL;
-  priv->create_mipmaps = TRUE;
+  priv->create_mipmaps = FALSE;
   priv->is_y_inverted = TRUE;
 }
 
-- 
2.14.3


